catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeColorByNAction.kt

```diff
@@ -20,38 +20,27 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-package org.catrobat.catroid.content.actions;
-
-import android.util.Log;
-
-import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
-
-import org.catrobat.catroid.content.Scope;
-import org.catrobat.catroid.formulaeditor.Formula;
-import org.catrobat.catroid.formulaeditor.InterpretationException;
-
-public class ChangeColorByNAction extends TemporalAction {
-
-	private static final String TAG = ChangeColorByNAction.class.getSimpleName();
-
-	private Scope scope;
-	private Formula color;
-
-	protected void update(float delta) {
-		try {
-			float eightBitColor = color == null ? 25 : color.interpretFloat(scope);
-			scope.getSprite().look.changeColorInUserInterfaceDimensionUnit(eightBitColor);
-		} catch (InterpretationException interpretationException) {
-			Log.d(TAG, "		Formula interpretation for this specific Brick failed.", interpretationException);
-		}
-	}
-
-	public void setColor(Formula color) {
-		this.color = color;
-	}
-
-	public void setScope(Scope scope) {
-		this.scope = scope;
-	}
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+
+private const val DEFAULT_COLOR = 25f
+
+class ChangeColorByNAction : TemporalAction() {
+    var scope: Scope? = null
+    var color: Formula? = null
+
+    override fun update(delta: Float) {
+        val notNullScope = scope ?: return
+        try {
+            val newColor: Float = color?.interpretFloat(scope) ?: DEFAULT_COLOR
```

Comment:
Please use the notNullScope.
You could have used [?.let {](https://discuss.kotlinlang.org/t/let-vs-if-not-null/3542), but your approach is also ok :)

Reply:
thanks for pointing this out. fixed it